compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEControlCodeToTransaction.cpp

```diff
@@ -223,33 +181,22 @@ LogicalResult convertOp(AMDAIE::NpuWriteBdOp op, TransactionBuilder &builder) {
   uint32_t col = op.getCol();
   uint32_t row = op.getRow();
   uint32_t bdId = op.getBdId();
-  uint32_t colShift = builder.deviceModel.getColumnShift();
-  uint32_t rowShift = builder.deviceModel.getRowShift();
-  uint32_t bdAddr =
-      (col << colShift) | (row << rowShift) | (0x1D000 + bdId * 0x20);
   ArrayRef<int32_t> sizes = op.getSizes();
   ArrayRef<int32_t> strides = op.getStrides();
   if (sizes.size() != 3) return op.emitOpError() << "expected 3 sizes";
   if (strides.size() != 3) return op.emitOpError() << "expected 3 strides";
-  uint32_t d0Size = sizes[sizes.size() - 1];
-  uint32_t d1Size = sizes[sizes.size() - 2];
-  // Strides and iteration_size are encoded as `actual - 1`, but `0` should stay
-  // `0` as it's not supported;
-  uint32_t d0Stride =
-      std::max((int64_t)strides[strides.size() - 1] - 1, (int64_t)0);
-  uint32_t d1Stride =
-      std::max((int64_t)strides[strides.size() - 2] - 1, (int64_t)0);
-  uint32_t d2Stride =
-      std::max((int64_t)strides[strides.size() - 3] - 1, (int64_t)0);
-  uint32_t iterationSize =
-      std::max((int64_t)op.getIterationSize() - 1, (int64_t)0);
+  // Strides and iteration_size will be encoded as `actual - 1`, so we need to
+  // ensure they are at least 1.
+  SmallVector<int32_t> stridesNew(strides.begin(), strides.end());
```

Comment:
You could perform the copy on `strides` directly to avoid having both `strides` and `stridesNew` around, for example:

```
SmallVector<int32_t> strides(op.getStrides());
```